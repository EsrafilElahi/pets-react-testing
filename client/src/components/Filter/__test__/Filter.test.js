import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Filters from '../Filter';


describe("Filter", () => {
  // render
  beforeEach(() => {
    render(<Filters filters={{}} setFilters={() => { }} />);
  });

  test("snapshot test", () => {
    const { asFragment } = render(<Filters filters={{}} setFilters={() => { }} />);
    expect(asFragment()).toMatchSnapshot();
  })

  test("should be able change value of fouvorite select", () => {
    // find element
    const favouriteSelect = screen.getByLabelText(/favourite/i);

    // assertion any
    expect(favouriteSelect).toHaveValue('any');

    // change favoured
    userEvent.selectOptions(favouriteSelect, "favoured");

    // assertion favoured
    expect(favouriteSelect).toHaveValue('favoured');

    // change not favoured
    userEvent.selectOptions(favouriteSelect, "not favoured");

    // assertion not favoured
    expect(favouriteSelect).toHaveValue('not favoured');
  });

  test("should be able change value of gender select", () => {
    // find element
    const genderSelect = screen.getByLabelText(/gender/i);

    // assertion any
    expect(genderSelect).toHaveValue('any');

    // change male
    userEvent.selectOptions(genderSelect, "male");

    // assertion male
    expect(genderSelect).toHaveValue("male");

    // change female
    userEvent.selectOptions(genderSelect, "female");

    // assertion female
    expect(genderSelect).toHaveValue("female");
  });
});