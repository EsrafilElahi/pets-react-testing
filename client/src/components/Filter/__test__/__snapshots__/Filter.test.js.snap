// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter snapshot test 1`] = `
<DocumentFragment>
  <div
    class="pet-filter-container"
  >
    <div
      class="filter-container"
    >
      <label
        for="favourite"
      >
        Favourite
      </label>
      <select
        class="form-select"
        id="favourite"
        name="favourite"
      >
        <option
          value="any"
        >
          Any
        </option>
        <option
          value="favoured"
        >
          Favoured
        </option>
        <option
          value="not favoured"
        >
          Not Favoured
        </option>
      </select>
    </div>
    <div
      class="filter-container"
    >
      <label
        for="gender"
      >
        Gender
      </label>
      <select
        class="form-select"
        id="gender"
        name="gender"
      >
        <option
          value="any"
        >
          Any
        </option>
        <option
          value="male"
        >
          Male
        </option>
        <option
          value="female"
        >
          Female
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
